src/Donations/Repositories/DonationMetaRepository.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Give\Donations\Repositories;
+
+use Give\Framework\Database\DB;
+use Give\Framework\QueryBuilder\QueryBuilder;
+use Give\Framework\Support\Facades\Str;
+
+/**
+ * @unreleased
+ */
+class DonationMetaRepository
+{
+    /**
+     * @unreleased
+     * @return mixed|null
+     */
+    public function get(int $donationId, string $metaKey)
+    {
+        $query = $this->prepareQuery()
+            ->where("donation_id", $donationId)
+            ->where("meta_key", $metaKey)
+            ->get();
+
+        if (!$query) {
+            return null;
+        }
+
+        $value = $query->meta_value;
+
+        if (Str::isJson($value)) {
+            return json_decode($value, false);
+        }
+
+        return $value;
+    }
+
+    /**
+     * @unreleased
+     */
+    public function upsert(int $donationId, string $metaKey, $metaValue): void
```

Comment:
This is where I was debating adding a DB transaction as this can be used standalone but is also nested in the DonationRepository.  Open to suggestions